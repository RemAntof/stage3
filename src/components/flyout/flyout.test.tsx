import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import itemsSelectedReducer from './cardsSlice';
import Flyout from './flyout';
jest.mock('@services/exportToCsv/exportToCsv', () =>
  jest.fn()
);

const store = configureStore({
  reducer: {
    itemsSelected: itemsSelectedReducer,
  },
});

describe('Flyout Component', () => {
  test('matches snapshot when items are present', () => {
    store.dispatch({
      type: 'itemsSelected/addItem',
      payload: {
        uid: 'ANMA0000005771',
        name: 'Andorian redbat',
        earthAnimal: false,
        earthInsect: false,
        avian: false,
        canine: false,
        feline: false,
      },
    });

    const { asFragment } = render(
      <Provider store={store}>
        <Flyout />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('matches snapshot when no items are present', () => {
    store.dispatch({
      type: 'itemsSelected/removeAllItems',
    });

    const { asFragment } = render(
      <Provider store={store}>
        <Flyout />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
