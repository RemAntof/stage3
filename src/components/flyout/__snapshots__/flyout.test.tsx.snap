// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flyout Component matches snapshot when items are present 1`] = `
<DocumentFragment>
  <div
    class="flyoutContainer"
  >
    <div
      class="flyoutContent"
    >
      1
    </div>
    <button
      class="closeButton"
    >
      Remove All Items
    </button>
    <button
      class="closeButton"
    >
      Download
    </button>
  </div>
</DocumentFragment>
`;

exports[`Flyout Component matches snapshot when no items are present 1`] = `<DocumentFragment />`;
